src/zoid/venmo/component.jsx

```diff
@@ -0,0 +1,310 @@
+          value: getEnv,
+        },
+
+        sdkMeta: {
+          type: "string",
+          queryParam: true,
+          value: getSDKMeta,
+        },
+
+        nonce: {
+          type: "string",
+          required: false,
+          value: getCSPNonce,
+          allowDelegate: true,
+        },
+
+        createAuthCode: {
+          type: "function",
+          queryParam: "code",
+          required: false,
+          // $FlowFixMe
+          queryValue: ({ value }) => ZalgoPromise.try(value),
+          // $FlowFixMe
+          decorate: ({ value }) => memoize(value),
+        },
+
+        buyerCountry: {
+          type: "string",
+          queryParam: true,
+          required: false,
+          default: getBuyerCountry,
+        },
+
+        locale: {
+          type: "object",
+          queryParam: "locale.x",
+          allowDelegate: true,
+          queryValue: ({ value }) => `${value.lang}_${value.country}`,
+          value: getLocale,
+        },
+
+        createOrder: {
+          type: "function",
+          queryParam: "token",
+          alias: "payment",
+          // $FlowFixMe
+          queryValue: ({ value }) => ZalgoPromise.try(value),
+          decorate: ({ value }) => memoize(value),
+        },
+
+        xcomponent: {
+          type: "string",
+          queryParam: true,
+          value: () => "1",
+        },
+
+        version: {
+          type: "string",
+          queryParam: true,
+          value: getVersion,
+        },
+
+        commit: {
+          type: "boolean",
+          queryParam: true,
+          value: getCommit,
+        },
+
+        fundingSource: {
+          type: "string",
+          queryParam: true,
+          default: () => FUNDING.VENMO,
+        },
+
+        standaloneFundingSource: {
+          type: "string",
+          queryParam: true,
+          required: false,
+        },
+
+        branded: {
+          type: "boolean",
+          queryParam: true,
+          required: false,
+        },
+
+        enableFunding: {
+          type: "array",
+          queryParam: true,
+          required: false,
+        },
+
+        onApprove: {
+          type: "function",
+          alias: "onAuthorize",
+        },
+
+        onComplete: {
+          type: "function",
+          required: false,
+        },
+
+        onShippingChange: {
+          type: "function",
+          required: false,
+        },
+
+        onShippingAddressChange: {
+          type: "function",
+          required: false,
+        },
+
+        onShippingOptionsChange: {
+          type: "function",
+          required: false,
+        },
+
+        clientMetadataID: {
+          type: "string",
+          required: false,
+          default: getClientMetadataID,
+          queryParam: "client-metadata-id",
+        },
+
+        onAuth: {
+          type: "function",
+          required: false,
+          trustedDomains: ["https://www.paypal.com", "https://venmo.com"],
```

Comment:
Out of curiosity, how's this going to work? I was originally assuming it would be something like `isTrustedDomain: isPayPalTrustedDomain()`.

Reply:
@nbierdeman How it works is one answer but it does cause me to think about the allowed types here. The `isPayPalTrustedDomain()` is a convenience function that uses RegEx to compare the trusted domain regex and should be used here. It is very possible to call each regex individually as well. The issue I am seeing here though is that the `trustedDomains` prop should verify against a regex rather than a string: `prop.trustedDomains.includes(getDomain(window))`. This will give us flexibility across environments. I'll make that change in void. Thanks for catching!